doc: remove security reporting examples

Trott opened the pull request:
Keep things concise and tell people to report stuff if in doubt.

This may be controversial (although I hope not). In my opinion, the level of detail provided seems  to be geared towards discouraging reports by making things seem unnecessarily complicated or nuanced.
<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

Trott commented:
@nodejs/security-wg @nodejs/security-triage

lpinca commented:
I think having past examples is good.

jasnell commented:
I think having examples is good but that current text and the way it's presented could be improved. Not blocking this change tho

sam-github commented:
The current text is intended to discourage *invalid* vulnerability reports, and can be pointed to after the fact to explain to people why their report was rejected. Recall that there has been a history of node vulnerability reports along the lines of "anyone can spawn a child process!", and that many node developers come from the browser and are suprised by the seeming vulnerability of node.

@nodejs/security The folks who do the initial work of triaging incoming vuln reports should weigh in on this.

@Trott The people who report vulnerabilities are generally comfortable with nuance and complexity - have you heard of anyone being daunted by this? Notice that one of the lines you *removed* was:

> When in doubt, please do send us a report nonetheless.

Trott commented:
> @Trott The people who report vulnerabilities are generally comfortable with nuance and complexity - have you heard of anyone being daunted by this? Notice that one of the lines you removed was:

@sam-github I removed one occurrence of that. That text appears twice. I removed all the text between the two occurrences. Saying it twice in a row seemed like overkill.

Trott commented:
> @nodejs/security The folks who do the initial work of triaging incoming vuln reports should weigh in on this.

@sam-github That would actually be @nodejs/security-triage. I pinged that team in my very first comment.

Trott commented:
> The current text is intended to discourage invalid vulnerability reports, and can be pointed to after the fact to explain to people why their report was rejected. 

@sam-github I think it (probably) doesn't do much of anything to discourage invalid reports, and is primarily there for your second reason, which is that it's something for us to point to after the fact. That could be a valid reason to have it, although I'm not a fan of text that serves our needs rather than the reader's needs. But if we need to have that text and there's not an obvious better place for it, then README it is....

By the way, I'm responding to you because your comments are GOOD, not because I take particular issue with them in ways that I don't with others. Thanks for the thoughtful commentary.

Trott commented:
Closing. Will try again with something more limited in change scope.
